go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -104,51 +104,40 @@ func (oc *Controller) syncEgressFirewall(egressFirewalls []interface{}) {
 // Upon failure, it may be invoked multiple times in order to avoid a pod restart.
 func (oc *Controller) syncEgressFirewallRetriable(egressFirewalls []interface{}) error {
 	// Lookup all ACLs used for egress Firewalls
-	egressFirewallACLs, err := libovsdbops.FindACLsByPriorityRange(oc.nbClient, types.MinimumReservedEgressFirewallPriority, types.EgressFirewallStartPriority)
+	aclPred := func(item *nbdb.ACL) bool {
+		return item.Priority >= types.MinimumReservedEgressFirewallPriority && item.Priority <= types.EgressFirewallStartPriority
+	}
+	egressFirewallACLs, err := libovsdbops.FindACLsWithPredicate(oc.nbClient, aclPred)
 	if err != nil {
 		return fmt.Errorf("unable to list egress firewall ACLs, cannot cleanup old stale data, err: %v", err)
 	}
 
 	if config.Gateway.Mode == config.GatewayModeShared {
 		// Mode is shared gateway mode, make sure to delete all egfw ACLs on the node switches
 		if len(egressFirewallACLs) != 0 {
-			err = libovsdbops.RemoveACLsFromNodeSwitches(oc.nbClient, egressFirewallACLs)
+			err = libovsdbops.RemoveACLsFromNodeSwitches(oc.nbClient, egressFirewallACLs...)
 			if err != nil {
 				return fmt.Errorf("failed to remove reject acl from node logical switches: %v", err)
 			}
 		}
 	} else if config.Gateway.Mode == config.GatewayModeLocal {
 		// Mode is local gateway mode, make sure to delete all egfw ACLs on the join switches
 		if len(egressFirewallACLs) != 0 {
-			err = libovsdbops.RemoveACLsFromJoinSwitch(oc.nbClient, egressFirewallACLs)
+			err = libovsdbops.RemoveACLsFromJoinSwitch(oc.nbClient, egressFirewallACLs...)
 			if err != nil {
 				return fmt.Errorf("failed to remove reject acl from node logical switches: %v", err)
 			}
 		}
 	}
 	// update the direction of each egressFirewallACL if needed
-	if len(egressFirewallACLs) != 0 {
-		opModels := []libovsdbops.OperationModel{}
-		for i := range egressFirewallACLs {
-			egressFirewallACL := egressFirewallACLs[i]
-			egressFirewallACL.Direction = types.DirectionToLPort
-			aclName := ""
-			if egressFirewallACL.Name != nil {
-				aclName = *egressFirewallACL.Name
-			}
-			opModels = append(opModels, libovsdbops.OperationModel{
-				Name:  aclName,
-				Model: &egressFirewallACL,
-				OnModelUpdates: []interface{}{
-					&egressFirewallACL.Direction,
-				},
-				ErrNotFound: true,
-			})
-		}
-		if _, err := oc.modelClient.CreateOrUpdate(opModels...); err != nil {
+	for i := range egressFirewallACLs {
```

Comment:
That goes a bit against a rule I set above in the PR message to avoid exploding libovsdbops with too many methods:
`Values to be updated to be provided through Model properties`
The current approach is a generic approach I took throughout all this PR. I am not sure of your proposal as a generic approach because you are assuming a bunch of items need to be updated to the same value. As an ad-hoc method it would only be useful here and we are not actually looping twice given that we are passing a single ACL to `UpdateACLsDirection`.
In any case, I would rather remove from libovsdbops API the ability to handle multiple items in a single call but it will be easier to do once we have level-driven controllers and a single transaction per event.
I would prefer to just use `CreateOrUpdateACLs` here instead of `UpdateACLsDirection` but I didn't want to take it that far on this PR.
All in all, I understand you suggestion but overall I would scope it out of the PR.

Reply:
ack sg!